test.disabled/ejabberd_tests/tests/sm_SUITE.erl

```diff
@@ -880,7 +902,14 @@ given_fresh_spec(Config, User) ->
 
 given_fresh_user(Config, UserName) ->
     Spec = given_fresh_spec(Config, UserName),
-    given_fresh_user_with_spec(Spec).
+    case proplists:is_defined(connection_steps, Config) of
+        true ->
+            Spec2 = [{connection_steps, proplists:get_value(connection_steps, Config)} | Spec];
```

Comment:
Should be 

```
Spec2 = case ... end
```

assigning inside case is some bad practice (most likely, banned in our coding standards).